src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsGeofencingZoneMapper.java

```diff
@@ -0,0 +1,65 @@
+package org.opentripplanner.updater.vehicle_rental.datasources;
+
+import com.google.common.hash.Hashing;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import org.entur.gbfs.v2_2.geofencing_zones.GBFSGeofencingZones;
+import org.locationtech.jts.geom.Geometry;
+import org.opentripplanner.framework.geometry.GeometryUtils;
+import org.opentripplanner.framework.geometry.UnsupportedGeometryException;
+import org.opentripplanner.routing.vehicle_rental.GeofencingZone;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GbfsGeofencingZoneMapper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GeofencingZone.class);
+
+  private final String systemId;
+
+  public GbfsGeofencingZoneMapper(String systemId) {
+    this.systemId = systemId;
+  }
+
+  public List<GeofencingZone> mapRentalVehicleType(GBFSGeofencingZones input) {
+    return input
+      .getData()
+      .getGeofencingZones()
+      .getFeatures()
+      .stream()
+      .map(f -> {
+        try {
+          var g = GeometryUtils.convertGeoJsonToJtsGeometry(f.getGeometry());
+          var name = Objects.requireNonNullElseGet(
+            f.getProperties().getName(),
+            () -> fallbackId(g)
+          );
+          var dropOffBanned = !f.getProperties().getRules().get(0).getRideAllowed();
+          var passThroughBanned = !f.getProperties().getRules().get(0).getRideThroughAllowed();
+          return new GeofencingZone(
+            new FeedScopedId(systemId, name),
+            g,
+            dropOffBanned,
+            passThroughBanned
+          );
+        } catch (UnsupportedGeometryException e) {
+          LOG.error("Could not convert geofencing zone", e);
+          return null;
+        }
+      })
```

Comment:
Extract this into a separate mapper method

Reply:
Done in a54c0d34c93564d7eca17b3c56c0deb27528177f

Reply:
Isn't this what this method is? Should I split it up into several smaller ones?

Reply:
I mean we could have the lambda on rows 32-51, which handles a single area, be its own method, and use a method call instead in `Stream#map`.

Reply:
Good point. Done in db609b6258b4bf1257a0b5a02cd1e877d92e9a76